src/constants.rs

```diff
@@ -59,1279 +59,157 @@
-pub const CONTROLLER_DOWNGRADE_SAFEMODE_THRESHOLD: u32 = 5000;
-pub const CONTROLLER_CLAIM_DOWNGRADE: u32 = 300;
-pub const CONTROLLER_RESERVE: u32 = 1;
-pub const CONTROLLER_RESERVE_MAX: u32 = 5000;
-pub const CONTROLLER_MAX_UPGRADE_PER_TICK: u32 = 15;
-pub const CONTROLLER_ATTACK_BLOCKED_UPGRADE: u32 = 1000;
-pub const CONTROLLER_NUKE_BLOCKED_UPGRADE: u32 = 200;
-
-pub const SAFE_MODE_DURATION: u32 = 20_000;
-pub const SAFE_MODE_COOLDOWN: u32 = 50_000;
-pub const SAFE_MODE_COST: u32 = 1000;
-
-pub const TOWER_HITS: u32 = 3000;
-pub const TOWER_CAPACITY: u32 = 1000;
-pub const TOWER_ENERGY_COST: u32 = 10;
-pub const TOWER_POWER_ATTACK: u32 = 600;
-pub const TOWER_POWER_HEAL: u32 = 400;
-pub const TOWER_POWER_REPAIR: u32 = 800;
-pub const TOWER_OPTIMAL_RANGE: u32 = 5;
-pub const TOWER_FALLOFF_RANGE: u32 = 20;
-pub const TOWER_FALLOFF: f32 = 0.75;
-
-pub const OBSERVER_HITS: u32 = 500;
-pub const OBSERVER_RANGE: u32 = 10;
-
-pub const POWER_BANK_HITS: u32 = 2_000_000;
-pub const POWER_BANK_CAPACITY_MAX: u32 = 5000;
-pub const POWER_BANK_CAPACITY_MIN: u32 = 500;
-pub const POWER_BANK_CAPACITY_CRIT: f32 = 0.3;
-pub const POWER_BANK_DECAY: u32 = 5000;
-pub const POWER_BANK_HIT_BACK: f32 = 0.5;
-
-pub const POWER_SPAWN_HITS: u32 = 1;
-pub const POWER_SPAWN_ENERGY_CAPACITY: u32 = 5000;
-pub const POWER_SPAWN_POWER_CAPACITY: u32 = 100;
-pub const POWER_SPAWN_ENERGY_RATIO: u32 = 50;
-
-pub const EXTRACTOR_HITS: u32 = 500;
-pub const EXTRACTOR_COOLDOWN: u32 = 5;
-
-pub const LAB_HITS: u32 = 500;
-pub const LAB_MINERAL_CAPACITY: u32 = 3000;
-pub const LAB_ENERGY_CAPACITY: u32 = 2000;
-pub const LAB_BOOST_ENERGY: u32 = 20;
-pub const LAB_BOOST_MINERAL: u32 = 30;
-pub const LAB_UNBOOST_ENERGY: u32 = 0;
-pub const LAB_UNBOOST_MINERAL: u32 = 15;
-
-pub const LAB_REACTION_AMOUNT: u32 = 5;
-
-pub const GCL_POW: f32 = 2.4;
-pub const GCL_MULTIPLY: u32 = 1_000_000;
-pub const GCL_NOVICE: u32 = 3;
-
-pub const TERRAIN_MASK_WALL: u8 = 1;
-pub const TERRAIN_MASK_SWAMP: u8 = 2;
-pub const TERRAIN_MASK_LAVA: u8 = 4;
-
-pub const MAX_CONSTRUCTION_SITES: u32 = 100;
-pub const MAX_CREEP_SIZE: u32 = 50;
-
-pub const MINERAL_REGEN_TIME: u32 = 50_000;
-
-/// Translates the `MINERAL_MIN_AMOUNT` constant.
-///
-/// Currently always returns 35_000 for all minerals...
-#[inline]
-pub fn mineral_min_amount(mineral: ResourceType) -> Option<u32> {
-    match mineral {
-        ResourceType::Hydrogen => Some(35_000),
-        ResourceType::Oxygen => Some(35_000),
-        ResourceType::Lemergium => Some(35_000),
-        ResourceType::Keanium => Some(35_000),
-        ResourceType::Zynthium => Some(35_000),
-        ResourceType::Utrium => Some(35_000),
-        ResourceType::Catalyst => Some(35_000),
-        _ => None,
-    }
+/// Re-export of all constants related to sources.
+pub mod source {
+    pub use super::numbers::{
+        ENERGY_REGEN_TIME, INVADERS_ENERGY_GOAL, SOURCE_ENERGY_CAPACITY,
+        SOURCE_ENERGY_KEEPER_CAPACITY, SOURCE_ENERGY_NEUTRAL_CAPACITY,
```

Comment:
Might want to add `ENERGY_DECAY` here.

Reply:
I _believe_ that constant is more related to how dropped resources behave than `Source`s. Unless you'd propose merging the two modules? I can get behind that.

Reply:
Well, beside fights, I expect to care about dropped energy when dealing with sources... Just a thought.

Reply:
Ah, that's a good point. I'll stick it in here too!